sailor/assetcentral/group.py

```diff
@@ -0,0 +1,219 @@
+"""
+Group module can be used to retrieve Groups information from AssetCentral.
+
+Classes are provided for individual Groups as well as groups of Groups (GroupSet).
+"""
+
+from functools import cached_property
+import warnings
+
+from .utils import AssetcentralEntity, ResultSet, _ac_application_url, _fetch_data, _apply_filters_post_request, \
+    _add_properties
+from .constants import VIEW_GROUPS
+from .equipment import find_equipment, EquipmentSet
+from .location import find_locations, LocationSet
+from .model import find_models, ModelSet
+
+
+@_add_properties
+class Group(AssetcentralEntity):
+    """AssetCentral Location Object."""
+
+    @classmethod
+    def get_property_mapping(cls):
+        """Return a mapping from assetcentral terminology to our terminology."""
+        return {
+            'id': ('id', None, None, None),
+            'name': ('displayId', None, None, None),
+            'group_type': ('groupTypeCode', None, None, None),
+            'short_description': ('shortDescription', None, None, None),
+            'risk_value': ('riskValue', None, None, None),
+        }
+
+    @cached_property
+    def _members_raw(self):
+        endpoint_url = _ac_application_url() + VIEW_GROUPS + f'/{self.id}/businessobjects'
+        object_list = _fetch_data(endpoint_url)
+        return object_list
+
+    def _generic_get_members(self, business_object_type, set_class, find_function, extended_filters, **kwargs):
+        member_name = set_class._element_type.__name__
+
+        if kwargs.get('id'):
+            raise RuntimeError(f'Can not specify `id` when retrieving "{member_name}" from a group.')
+
+        kwargs['id'] = [item['businessObjectId'] for item in self._members_raw
+                        if item['businessObjectType'] == business_object_type]
+
+        if not kwargs['id']:
+            warnings.warn(f'There are no "{member_name}" in this group!')
```

Comment:
You could use the DataNotFoundWarning we have for cases like this. Might not be completely accurate though. Give it a thought.